eth/peerset.go

```diff
@@ -402,7 +404,15 @@ func (ps *peerSet) waitBscExtension(peer *eth.Peer) (*bsc.Peer, error) {
 		return peer, nil
 
 	case <-time.After(extensionWaitTimeout):
-		ps.lock.Lock()
+		// once timeout is reached, we won't wait for the peer any more, bsc protocol won't be enabled on it.
+		log.Warn("waitBscExtension", "peer", id, "bsc protocol timeout, it won't be enabled.")
+		for {
+			// it is potential deadlock, so we use TryLock to avoid it.
+			if ps.lock.TryLock() {
```

Comment:
still deadlock:
    wait <- nil, blocked because no one to accept nil from channel
    ps.lock.TryLock() always return false at the same time

Reply:
yes, forget to clean the channel... will update